test/unit/lib/plugins/aws/package/compile/functions.test.js

```diff
@@ -2397,120 +1521,189 @@ describe('lib/plugins/aws/package/compile/functions/index.test.js', () => {
         },
       });
       cfResources = cfTemplate.Resources;
+      cfOutputs = cfTemplate.Outputs;
       naming = awsNaming;
       serviceConfig = fixtureData.serviceConfig;
-      iamRolePolicyStatements =
-        cfResources.IamRoleLambdaExecution.Properties.Policies[0].PolicyDocument.Statement;
     });
 
-    it.skip('TODO: should support `package.artifact`', async () => {
-      // Replacement for:
-      // https://github.com/serverless/serverless/blob/d8527d8b57e7e5f0b94ba704d9f53adb34298d99/lib/plugins/aws/package/compile/functions/index.test.js#L149-L168
+    it('should support `package.artifact`', async () => {
+      const {
+        Code: { S3Key },
+      } = cfResources[naming.getLambdaLogicalId('other')].Properties;
+
+      expect(S3Key.endsWith('testServiceArtifact.zip')).to.be.true;
     });
 
-    it.skip('TODO: should ignore `package.artifact` when packaging individually', () => {
-      // Replacement for
-      // https://github.com/serverless/serverless/blob/d8527d8b57e7e5f0b94ba704d9f53adb34298d99/lib/plugins/aws/package/compile/functions/index.test.js#L190-L211
+    it('should ignore `package.artifact` when packaging individually', () => {
```

Comment:
Let's rename this test case to: _should prefer `functions[].package.artifact` over `service.artifact`_

(when naming that test case, I assumed that this requires `package.individually: true`, and that it's what we test here, but that's not the case)